main/streams/plain_wrapper.c

```diff
@@ -53,12 +53,16 @@ extern int php_get_uid_by_name(const char *name, uid_t *uid);
 extern int php_get_gid_by_name(const char *name, gid_t *gid);
 #endif
 
-#if defined(PHP_WIN32)
+#if defined(PHP_WIN32) || defined(__APPLE__) || defined(__MACH__) || defined(BSD) || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
```

Comment:
Is it possible to do a configure-time check for this?

Reply:
I'm not really sure if Solaris can handle it either (can't be bother to check) so safest option should be to just do `#ifdef __linux__` and invert the logic...

Reply:
It would probably be salient to also add a comment explaining why then. (the BSD and Windows cause should explain pretty much all cases).

(FWIW, AIX does seem to allow reads as big as `off_t` on 64-bit systems...)